test/fluree/server/integration/policy_test.clj

```diff
@@ -93,33 +96,83 @@
-                          "ledger"   ledger-name
-                          "insert"   [{"id" "ex:bob"}
-                                      "ex:secret" "bob's new secret"]
-                          "opts"     {"role" "ex:userRole"
-                                      "did"  alice-did}})
-                       :headers json-headers}
-              txn-res (api-post :transact txn-req)]
-          (is (not= 200 (:status txn-res))
-              (str "transaction policy opts should have prevented modification, instead response was: " (pr-str txn-res)))
-          (let [query-req {:body
-                           (json/write-value-as-string
-                             {"@context" test-system/default-context
-                              "from"     ledger-name
-                              "history"  "ex:bob"
-                              "t"        {"from" 1}
-                              "opts"     {"role" "ex:userRole"
-                                          "did"  alice-did}})
-                           :headers json-headers}
-                query-res (api-post :history query-req)]
-            (is (= 200 (:status query-res))
-                (str "History query response was: " (pr-str query-res)))
-            (is (= [{"id" "ex:bob", "type" "ex:User"}]
-                   (-> query-res :body json/read-value first (get "f:assert")))
-                "policy opts should have prevented seeing bob's secret")))))))
+        (testing "plain requests"
+          (let [txn-req {:body
+                         (json/write-value-as-string
+                           {"@context" ["https://ns.flur.ee" test-system/default-context]
+                            "ledger"   ledger-name
+                            "insert"   [{"id" "ex:bob"}
+                                        "ex:secret" "bob's new secret"]
+                            "opts"     {"role" "ex:userRole"
+                                        "did"  alice-did}})
+                         :headers json-headers}
+                txn-res (api-post :transact txn-req)]
+            (is (not= 200 (:status txn-res))
+                (str "transaction policy opts should have prevented modification, instead response was: " (pr-str txn-res)))
+            (let [query-req {:body
+                             (json/write-value-as-string
+                               {"@context" test-system/default-context
+                                "from"     ledger-name
+                                "history"  "ex:bob"
+                                "t"        {"from" 1}
+                                "opts"     {"role" "ex:userRole"
+                                            "did"  alice-did}})
+                             :headers json-headers}
+                  query-res (api-post :history query-req)]
+              (is (= 200 (:status query-res))
+                  (str "History query response was: " (pr-str query-res)))
```

Comment:
Good point, I had just copied what was there. I've updated them now.

Reply:
I don't think I agree with this. The second string arg to `is` has been most useful to me when it shows what did happen w/ more info than the test failure alone. It's fine if it also says what should have happened, but that's already pretty obvious from the test itself (at least it should be). You only ever see it when the test fails, so it's a nice way to give some more details about the failure. So I would say the exact opposite: figuring out what should happen is usually obvious (the test failure report says "I expected X but got Y"), but figuring out what did happen isn't always. And this is a good place to provide that info.

Reply:
For example:

If this test is changed to document what should have happened, then on failure you'll see (roughly):

```
Expected: 200
Actual: 400
Response is successful
```

😕

Reply:
> It's most important that they explain what should happen

This is also my understanding of the `is` string convention. But to avoid redundancy with the diff you get anyway, I think they're best used to give some kind of context/explanation for what was expected (and sometimes why) in _human_ terms. 

I agree that a string of `"Response should be successful"` is not useful. But something like `"Policy should have allowed XYZ, because ABC"`. Which in this case may not seem the most illustrative either?, but that's the idea as I understand it.

Reply:
I don't feel strongly one way or another. I think putting what went wrong in the `is` 3rd arg makes reading the test source code weirder. I also think it makes the test output easier to read. 

However, I'd like to get this merged so Andrew can incorporate these changes in our docs. Would it be acceptable to make a ticket to decide test conventions so we can apply them consistently? I know we have a ton of tests that describe the expected behavior in `is`, so I don't think we necessarily need to block this PR on that decision.

Reply:
I agree with that. I have some thoughts on this stuff that we can discuss in a different forum, but we shouldn't hold back this pr one way or another.

Reply:
To be clear, I'm not advocating for putting "what went wrong" in the second `is` arg. I'm advocating for putting any additional context, etc. that will help determine why the test failed.

And yes, we can move discussion to a future ticket. I think there is currently a mix of approaches here in our source code and unfortunately [the example in the official docs](https://clojuredocs.org/clojure.test/is) is basically useless for determining the intent of the arg.